.travis.yml

```diff
@@ -8,4 +11,7 @@ cache:
     - $HOME/.m2 # Maven repository
 script:
   - sbt -Dhive.jdbc.version=upstream assembly
+  - cd ./testsuit-extras/test-http-server
+  - python3 -m http.server 9000 & # start the server for http related tests
```

Comment:
As the number of tricky things we need to test grows the approach of setting up the environment *before* running all tests will quickly become unsustainable, so I'm a big fan of tests that fully take care of their own setup.

In your case I think it would be better to:
- use `DatasetFactory` to create mock dataset
- dump it into a temporary yaml file
- spawn a docker container with an http server and mount this file into it (we already have a nice docker process wrapper for this) (you should be able to find a good 3rd party image for this task so we don't have to maintain our own)
- wait for it to start by trying to connect to it (we already have this for Livy startup, so can be generalized and reused)
- execute the rest of the test and clean up

Reply:
Sounds a bit over complicated(third-party image, docker, mounting files to it, not to mention potential security risks from somebody-else's images) compared to single call to python's standard library.

Reply:
I wouldn't call it a "single call" - this is something that every developer will have to do every time they run tests, and something that will need to be documented. `python3` is also not a portable way to launch python and will not work in my environment for example, so it really brings a lot of unnecessary complexity. I'd rather hide this complexity from people within a test.

You can still use python's http server for this, but starting it via docker from one of the [official slim images](https://hub.docker.com/_/python) is simpler than requiring people to install python in a certain way. Don't think there's any practical security concerns with those images - at work we use them both for local dev and for prod deployments.

So yes, it's more work to do within a test but it's more productive in the long run, it scales well to testing different configurations, and we already have all necessary building blocks to make it simple to implement.